### PR TITLE
Support for Windows vm agent

### PR DESCRIPTION
**Pull request by Anandkumar26**
- Create VM switch an interface that has default route
- Enable OVS extension on VM switch
- Move uplink and host interface into OVS
- Populate the interface cache, so that when external entity
is found, it updates the ovs configuration.
- Handle restart/reboot cases

Signed-off-by: Anand Kumar <kumaranand@vmware.com>

### CODE REVIEW

File: pkg/agent/util/net_windows.go [hunk 1]
```diff
@@ -903,7 +903,9 @@ func MoveIFConfigurations(ips []*net.IPNet, routes []interface{}, mac net.Hardwa
 			GatewayAddress:    rt.GatewayAddress,
 			RouteMetric:       rt.RouteMetric,
 		}
-		if err := NewNetRoute(&newRt); err != nil {
+		if err := ReplaceNetRoute(&newRt); err != nil {
+			//if err := NewNetRoute(&newRt); err != nil {
```
Comment (wenyingd) [inline pkg/agent/util/net_windows.go:907]: Remove this line?

File: pkg/agent/externalnode/external_entity_controller.go [hunk 2]
```diff
@@ -421,6 +421,7 @@ func (c *ExternalEntityController) getHostInterfaceNameByEndpoint(ep v1alpha2.En
 		}
 		_, _, link, err := util.GetIPNetDeviceFromIP(ipFilter, sets.NewString())
 		if err == nil {
+			klog.InfoS("Using the interface", "link.Name", link.Name)
```
Comment (wenyingd) [inline pkg/agent/externalnode/external_entity_controller.go:424]: s/"link.Name"/"linkName"

File: pkg/agent/agent.go [hunk 3]
```diff
@@ -332,6 +334,7 @@ func (i *Initializer) initInterfaceStore() error {
 			}
 		}
 		if intf != nil {
+			klog.V(2).InfoS("Adding interface to cache", "InterfaceName", intf.InterfaceName)
```
Comment (wenyingd) [inline pkg/agent/agent.go:337]: s/InterfaceName/interfaceName

File: pkg/agent/util/net_windows.go (hunk 1 above)
Comment (Anandkumar26) [reply pkg/agent/util/net_windows.go:907]: Done

File: pkg/agent/agent.go (hunk 3 above)
Comment (Anandkumar26) [reply pkg/agent/agent.go]: Done

File: pkg/agent/externalnode/external_entity_controller.go (hunk 2 above)
Comment (Anandkumar26) [reply pkg/agent/externalnode/external_entity_controller.go]: Done

**Pull request merged**